autoscheduler/scraper/management/commands/scrape_courses.py

```diff
@@ -134,53 +146,120 @@ def parse_course(course_data):
         title = title[4:]
     credit_hours = course_data['creditHourLow']
 
+    # Parse the instructor, then send the returned Instructor model to parse_section
+    instructor_model = parse_instructor(course_data, instructors_set)
+    section_data = parse_section(course_data, instructor_model)
+
     # Save course only if it hasn't already been created
-    if course_id not in COURSES_SET:
+    if course_id not in courses_set:
         course_model = Course(id=course_id, dept=dept, course_num=course_number,
                               title=title, credit_hours=credit_hours, term=term_code)
-        course_model.save()
+        courses_set.add(course_id)
+        return (course_model, instructor_model, section_data)
 
-    COURSES_SET.add(course_id)
+    return (None, instructor_model, section_data)
 
-    # Parse the instructor, then send the returned Instructor model to parse_section
-    instructor_model = parse_instructor(course_data)
-    parse_section(course_data, instructor_model)
 
-def get_department_names(term_code: str) -> List[str]:
+def get_department_names(term: str) -> List[str]:
     """ Queries database for list of all departments """
-    return [dept.code for dept in Department.objects.filter(term=term_code)]
+    return ((dept.code, term) for dept in Department.objects.filter(term=term))
```

Comment:
Can't make a suggestion because there's a removed line, but we can only do 1 query by doing this and passing in terms as a set:
```python
def get_department_names(terms: List[str]) -> List[str]:
    """ Queries database for list of all departments """
    depts = (Department.objects.filter(term__in=terms).order_by('term')
                               .values('code', 'term'))
    grouped = groupby(depts, key=lambda x: x['term'])
    return ((dept['code'], term) for term, group in grouped for dept in group)
```
I ran the querying part of the code 20 times for a single term and all terms, here's how the times compare:
|           |  old   | new    |
|-----------|--------|--------|
| one term  | 0.0320 | 0.0302 |
| all terms | 0.8955 | 0.4021 |

Reply:
Yeah great point, will add